javascriptcore-macros/src/lib.rs

```diff
@@ -9,59 +25,82 @@ pub fn function_callback(_attributes: TokenStream, item: TokenStream) -> TokenSt
 
     quote! {
         unsafe extern "C" fn #function_name(
-            __raw_ctx: javascriptcore_sys::JSContextRef,
-            __raw_function: javascriptcore_sys::JSObjectRef,
-            __raw_this_object: javascriptcore_sys::JSObjectRef,
-            __raw_argument_count: usize,
-            __raw_arguments: *const javascriptcore_sys::JSValueRef,
-            __raw_exception: *mut javascriptcore_sys::JSValueRef,
+            raw_ctx: javascriptcore_sys::JSContextRef,
+            function: javascriptcore_sys::JSObjectRef,
+            this_object: javascriptcore_sys::JSObjectRef,
+            argument_count: usize,
+            arguments: *const javascriptcore_sys::JSValueRef,
+            exception: *mut javascriptcore_sys::JSValueRef,
         ) -> *const javascriptcore_sys::OpaqueJSValue {
-            use core::{mem, ptr, slice};
+            use core::{mem::ManuallyDrop, ptr, slice};
             use javascriptcore::{JSContext, JSObject, JSValue};
+            use javascriptcore_sys::JSValueRef;
 
-            let __ctx = JSContext::from_raw(__raw_ctx as *mut _);
-            let __function = JSObject::from_raw(__raw_ctx, __raw_function);
-            let __this_object = JSObject::from_raw(__raw_ctx, __raw_this_object);
+            // This should never happen, it's simply a paranoid precaution.
+            if raw_ctx.is_null() {
```

Comment:
I sort of wonder if this should be an assert, if it should never happen.

Reply:
That's indeed a good idea. Should I open a new PR?

Reply:
Sure! Should probably note in the docs somewhere that it will panic in that erroneous case.